### PR TITLE
[sw/silicon_creator] Constrain otp_read to aligned reads

### PR DESCRIPTION
**Pull request by jon-flatley**
Update the signature for otp_read as a follow-up to
https://github.com/lowRISC/opentitan/pull/7000/files#r652890537

### CODE REVIEW

File: sw/device/silicon_creator/lib/drivers/otp.c [hunk 1]
```diff
@@ -28,18 +28,9 @@ uint64_t otp_read64(uint32_t address) {
...
 }
 
-rom_error_t otp_read(uint32_t address, void *data, size_t len) {
-  // TODO Update to use alignment utility functions.
-  // https://github.com/lowRISC/opentitan/issues/6112
-  if (address % alignof(uint32_t) != 0 || len % sizeof(uint32_t) != 0) {
-    return kErrorOtpBadAlignment;
-  }
-
+void otp_read(uint32_t address, uint32_t *data, size_t len) {
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/otp.c:31]:
```suggestion
void otp_read(uint32_t address, uint32_t *data, size_t num_words) {
```

File: sw/device/silicon_creator/lib/drivers/otp.c [hunk 2]
```diff
@@ -28,18 +28,9 @@ uint64_t otp_read64(uint32_t address) {
...
   uint32_t reg_offset = OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + address;
-  for (size_t i = 0; i < len; i += sizeof(uint32_t)) {
-    uint32_t word = sec_mmio_read32(kBase + reg_offset + i);
-    memcpy(data + i, &word, sizeof(uint32_t));
+  for (size_t i = 0; i < len; ++i) {
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/otp.c:33]:
```suggestion
  for (size_t i = 0; i < num_words; ++i) {
```

File: sw/device/silicon_creator/lib/drivers/otp.h [hunk 3]
```diff
@@ -43,12 +43,9 @@ uint64_t otp_read64(uint32_t address);
  *
  * @param address The address to read from offset from the start of OTP memory.
  * @param data The output buffer of at least length `len`.
- * @param len The number of bytes to read from OTP.
- * @return `kErrorOtpBadAlignment` if `address` or `address` + `len` are
- * misaligned, `kErrorOk` otherwise.
+ * @param len The number of 32-bit words to read from OTP.
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/otp.h:46]: nit: `len` -> `num_words`. Also, the comment above should be updated (still mentions bytes and alignment).

File: sw/device/silicon_creator/lib/drivers/otp_unittest.cc [hunk 4]
```diff
@@ -64,20 +65,12 @@ TEST_F(OtpReadTest, ReadLenN) {
   }
 
   std::vector<uint32_t> arr(16);
-  EXPECT_EQ(otp_read(0, &arr[0], arr.size() * sizeof(uint32_t)), kErrorOk);
+  otp_read(0, &arr[0], arr.size());
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/otp_unittest.cc:68]: nit: WDYT about `std::array<uint32_t, 16> arr;` here and `std::array<uint32_t, 16> expected;` below?

```suggestion
  otp_read(0, arr.data(), arr.size());
```

**Review by alphan — APPROVED** [review]
Thanks @jon-flatley, just some nits.

File: sw/device/silicon_creator/lib/drivers/otp_unittest.cc (hunk 4 above)
Comment (jon-flatley) [reply sw/device/silicon_creator/lib/drivers/otp_unittest.cc:68]: I think that's a fine idea!

File: sw/device/silicon_creator/lib/drivers/otp_unittest.cc [hunk 5]
```diff
@@ -45,39 +48,31 @@ TEST_F(OtpReadTest, ReadLen32) {
...
 }
 
 TEST_F(OtpReadTest, ReadLen64) {
   EXPECT_SEC_READ32(mmio_, base_ + offset_, 0x0C0D0E0F);
   EXPECT_SEC_READ32(mmio_, base_ + offset_ + 4, 0x08090A0B);
 
-  uint64_t value = 0;
-  EXPECT_EQ(otp_read(0, &value, sizeof(value)), kErrorOk);
-  EXPECT_EQ(value, 0x08090A0B0C0D0E0F);
+  uint32_t buf[2];
```
Comment (mcy) [inline sw/device/silicon_creator/lib/drivers/otp_unittest.cc:59]: Maybe this should be std::array, too? =P

**Review by mcy — APPROVED** [review]

**Review by cfrantz — APPROVED** [review]

**Pull request merged**